### PR TITLE
feat(list): better loading placeholder

### PR DESCRIPTION
**Pull request by baptisteArno**
## Summary

Improve loading placeholder to avoid content shifting

## Type

- Enhancement

## Before

https://user-images.githubusercontent.com/16015833/191781289-0afc08c1-8cad-49ce-ac7f-aacd2b9fb782.mp4

## After

https://user-images.githubusercontent.com/16015833/191781355-5ddcff4b-dc20-4538-ae02-b24e24bf11e9.mp4

### CODE REVIEW

**Comment by chambo-e** [comment]
Should we remove the List variant in Placeholder ?

File: src/components/List/LoadingPlaceholder.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  columns,
+}: LoadingPlaceholderProps<DataType>) => {
+  const rows = useMemo(() => Array.from({ length: totalRows }), [totalRows])
+
+  return (
+    <>
+      {rows.map((_, index) => (
+        <Row id={index.toString()}>
+          {columns.map(column => (
+            <Cell>{column.label ? <Placeholder variant="line" /> : null}</Cell>
```
Comment (matthprost) [inline src/components/List/LoadingPlaceholder.tsx:22]: key might be missing as we do a map

**Review by matthprost — CHANGES_REQUESTED** [review]

**Comment by baptisteArno** [comment]
> Should we remove the List variant in Placeholder ?

We don't use it anywhere at the moment, so it's removable. Do we have things in mind for the `list` Placeholder variant? @matthprost

File: src/components/List/LoadingPlaceholder.tsx (hunk 1 above)
Comment (baptisteArno) [reply src/components/List/LoadingPlaceholder.tsx:22]: Should I set column index as the key? I think it's useless in that case

**Comment by matthprost** [comment]
> > Should we remove the List variant in Placeholder ?
> 
> We don't use it anywhere at the moment, so it's removable. Do we have things in mind for the `list` Placeholder variant? @matthprost

Variant list is used on others project so we might want to keep it :)

**Review by DorianMaliszewski — APPROVED** [review]

**Review by matthprost — APPROVED** [review]

**Pull request merged**